doc/user-guide/04-scenarios.adoc

```diff
@@ -67,6 +67,106 @@ a single confidential command to let the redirection also apply for 'set -x'.
 See the description in /usr/share/rear/conf/default.conf
 how to set variables for secret values in a confidential way.
 
+== Booting ReaR rescue initramfs on IBM Z (s390x) under z/VM
+
+There are two options how to boot the rescue environment on IBM Z (s390x) under
+z/VM created using the `OUTPUT=RAMDISK` setting.
+
+1. Using the `kexec` tool as described in link:../../usr/share/rear/conf/templates/RESULT_usage_RAMDISK.txt[RESULT_usage_RAMDISK.txt],
+   if you have access to a running Linux instance.
+
+2. Using the `zipl` tool for management of boot devices on s390/s390x.
+
+=== Booting ReaR rescue initramfs on IBM Z (s390x) under z/VM with `zipl`
+
+After creating the rescue image with `rear mkrescue`, move the kernel and
+initramfs to a directory (denoted by `REAR_DIRECTORY` in this example) on the
+target device to boot from.
```

Comment:
I know that this has been discussed between you and @jsmeix a lot, but I think the use of `REAR_DIRECTORY` (and also `BOOTMAP_INSTALL_DIRECTORY`) is confusing. Since you are giving an example and in real use one will most likely use an absolute path as the argument to `zipl`, I think it is better to provide a realistic example, rather than providing a placeholder that the user should replace. Also, I think you should then describe more clearly what is the relation of the directory to the disk. You say that the directory is on the target device to boot from, which is correct, but maybe it should be written in a more explicit way (the style of this document is the one of specific examples, not the one of a reference manual), like this:
"Mount a partition on the disk that you want to make bootable. Let's say that the disk is called `/dev/dasda`, the partition with the filesystem that will contain the kernel and initramfs is `/dev/dasda1` and we mount it on `/mnt/bootdisk` :
```
# mount /dev/dasda1 /mnt/bootdisk
```
Copy the kernel and initramfs there:
```
# cp <something, TBD> /mnt/bootdisk
```
"
and from this point use `/mnt/bootdisk` instead of `REAR_DIRECTORY` and `BOOTMAP_INSTALL_DIRECTORY` in the example.
WDYT?

Reply:
This part seems to be endlessly complicated
and all descriptions look confusing
as long as one does not understand
the IPL implementation details behind :-(

Now I am wondering how the IBM Z "initial bootloader"
(I don't know how that thingy is called in IBM Z speech,
 I mean what loads kernel and initrd and executes it)
accesses kernel and initrd on the boot disk.

With
```
zipl --target=BOOTMAP_INSTALL_DIRECTORY \
       --parameters='root=/dev/ram0 ro' \
       --image=REAR_DIRECTORY/kernel-HOSTNAME \
       --ramdisk=REAR_DIRECTORY/initramfs-HOSTNAME.img
```
it looks as if the "initial bootloader"
loads kernel and initrd via their provided
filesystem names i.e.
```
REAR_DIRECTORY/kernel-HOSTNAME
REAR_DIRECTORY/initramfs-HOSTNAME.img
```
but then
```
/mnt/bootdisk/kernel-HOSTNAME
/mnt/bootdisk/initramfs-HOSTNAME.img
```
would not exist at IPL time because
/mnt/bootdisk/ does not exist at IPL time
instead /kernel-HOSTNAME and /initramfs-HOSTNAME.img
exist in the filesystem on /dev/dasda1 at IPL time.

When
```
zipl --target=/mnt/bootdisk \
       --parameters='root=/dev/ram0 ro' \
       --image=/mnt/bootdisk/kernel-HOSTNAME \
       --ramdisk=/mnt/bootdisk/initramfs-HOSTNAME.img
```
works, it means the "initial bootloader"
accesses kernel and initrd not via the
filesystem on /dev/dasda1
but differently.

I guess the "initial bootloader"
accesses kernel and initrd as LILO did it
by accessing directly the hardware "sectors"
on /dev/dasda1 where kernel and initrd are stored.

When my guess is right, it also means that
each time when ReaR creates a new recovery system
(i.e. after each "rear mkrescue/mkbackup")
```
mount /dev/dasda1 /mnt/bootdisk

cp <something, TBD> /mnt/bootdisk

zipl --target=/mnt/bootdisk \
       --parameters='root=/dev/ram0 ro' \
       --image=/mnt/bootdisk/kernel-HOSTNAME \
       --ramdisk=/mnt/bootdisk/initramfs-HOSTNAME.img

umount /mnt/bootdisk
```
must be called.

I guess this is what
rear/finalize/Fedora/s390/660_install_zipl.sh
and
rear/finalize/SUSE_LINUX/s390/660_install_grub2_and_zipl.sh
implement (or try to implement).

Am I right or do I misunderstand things?

https://manpages.debian.org/jessie/s390-tools/zipl.8
is rather vague what the purpose of the 'bootmap' file is
but I guess in particular the hardware "sectors"
of kernel and initrd are stored in the 'bootmap' file.

By the way:
Right now I found in
https://manpages.debian.org/jessie/s390-tools/zipl.8
```
-a or --add-files
Copy all specified files to the bootmap file
instead of just referencing them.
This option allows specifying files in a boot configuration
which are not located on the target device.
```
This looks as if 'zipl -a' is a fail-safe method
to be able to still boot regardless what happened later
after the 'zipl' call with the kernel and initrd files
at their location in the filesystem on /dev/dasda1
i.e. changing /mnt/bootdisk/kernel-HOSTNAME
or /mnt/bootdisk/initramfs-HOSTNAME.img
won't destroy a boot mechanism with that files.

Reply:
I admit that I also don't know if zipl is more like 'lilo' or more like 'grub' and this difference indeed matters from the POV of an administrator.

Reply:
I've made the steps more explicit in 873884f.  Thank you both for the suggestions!